src/mam4xx/lin_strat_chem.hpp

```diff
@@ -0,0 +1,322 @@
+    //
+    const int ltrop, const Real chlorine_loading,
+    const Real psc_T, // PSC ozone loss T (K) threshold
+    const ColumnView &o3_vmr,
+    // diagnostic variables outputs
+    const ColumnView &do3_linoz, const ColumnView &do3_linoz_psc,
+    const ColumnView &ss_o3, const ColumnView &o3col_du_diag,
+    const ColumnView &o3clim_linoz_diag, const ColumnView &sza_degrees) {
+
+  Kokkos::parallel_for(
+      Kokkos::TeamThreadRange(team, ltrop), KOKKOS_LAMBDA(int kk) {
+        lin_strat_chem_solve_kk(
+            o3col(kk), temperature(kk), sza, pmid(kk), delta_t, rlats,
+            // ltrop, & !in
+            linoz_o3_clim(kk), linoz_t_clim(kk), linoz_o3col_clim(kk),
+            linoz_PmL_clim(kk), linoz_dPmL_dO3(kk),
+            linoz_dPmL_dT(kk), // in
+            linoz_dPmL_dO3col(kk),
+            linoz_cariolle_psc(kk), // in
+                                    //
+            chlorine_loading,
+            psc_T, // PSC ozone loss T (K) threshold
+            o3_vmr(kk),
+            // diagnostic variables outputs
+            do3_linoz(kk), do3_linoz_psc(kk), ss_o3(kk), o3col_du_diag(kk),
+            o3clim_linoz_diag(kk), sza_degrees(kk));
+      });
+
+} // lin_strat_chem_solve
+
+KOKKOS_INLINE_FUNCTION
+void lin_strat_sfcsink(const Real delta_t, const ColumnView &pdel, // in
+                       const ColumnView &o3l_vmr, const Real o3_sfc,
+                       const int o3_lbl, const Real o3_tau, Real &o3l_sfcsink) {
+
+  // @param[in] delta_t              timestep size [secs]
+  // @param[in] pdel(ncol,pver)      pressure delta about midpoints [Pa]
+
+  // !inten in-outs
+  // @param[inout]:: o3l_vmr(ncol ,pver)             ! ozone volume mixing ratio
+  // [vmr]
+
+  constexpr Real one = 1.0;
+  // BAD CONSTANT
+  constexpr Real mwo3 = 48.; // molecular weight O3
+  constexpr Real mwdry = haero::Constants::molec_weight_dry_air *
+                         1e3; //     ! molecular weight dry air ~ kg/kmole;//!
+                              //     molecular weight dry air
+  constexpr Real rgrav =
+      one / haero::Constants::gravity; // reciprocal of gravit
+
+  const Real efactor =
+      (one - haero::exp(-delta_t / o3_tau)); // !compute time scale factor
+  Real do3mass_icol = 0;
+  for (int kk = pver - 1; kk > pver - o3_lbl - 1; --kk) {
+    const Real mass = pdel(kk) * rgrav; //   air mass in kg/m2
+
+    const Real o3l_old = o3l_vmr(kk);              // vmr
+    const Real do3 = (o3_sfc - o3l_old) * efactor; // vmr
+    const Real o3l_new = o3l_old + do3;
+    do3mass_icol +=
+        do3 * mass * mwo3 / mwdry; //   loss in kg/m2 summed over boundary
+                                   //   layers within one time step
+    o3l_vmr(kk) = o3l_new;
+  }
+
+  // Two parameters are applied to Linoz O3 for surface sink, O3l is not coupled
+  // to real O3
+  constexpr Real KgtoTg = 1.0e-9; //
+  constexpr Real peryear =
+      86400. * 365.0; // ! to multiply to convert per second to per year
```

Comment:
Probably not relevant to the porting effort, but should leap years be accounted for? Does it matter in the context of climate?

Reply:
I also hate that name because it should like it should be the reciprocal of what it is.

Reply:
> Probably not relevant to the porting effort, but should leap years be accounted for? Does it matter in the context of climate?

@brigaudet any comment on this?

Reply:
> I also hate that name because it should like it should be the reciprocal of what it is.

do you mean rename this variable to `perseconds` or `seconds_per_year`.

Reply:
> > Probably not relevant to the porting effort, but should leap years be accounted for? Does it matter in the context of climate?
> 
> @brigaudet any comment on this?

I may not be qualified to address this in terms of specific details of E3SM climate modeling diagnostics.  But if pressed, what I would say is:  this quantity is just a depletion rate over the interval of a model timestep, just converted from something with unit 'seconds'  to unit  'year' in the denominator.  Seconds are a fixed interval of length whereas 'year' is not because of leap year (or technically, seconds are not quite fixed in length but much more fixed than years are).  So I would argue it's understood here that 'year' can be defined as 'period of 365 days', since one is not actually computing a depletion integrated over a whole actual year.  Having said that, someone could argue that this constant should be '86400. * 365.2425....' in order to get the most correct long-term annual statistics.